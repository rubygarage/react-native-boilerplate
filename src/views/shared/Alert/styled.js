import styled from 'styled-components/native';

import Icon from 'views/shared/Icon';

const Wrapper = styled.View`
  position: relative;
  flex-direction: row;
  margin-bottom: 8;
  padding-top: 12;
  padding-right: ${({ onCloseClick }) => (onCloseClick ? 48 : 12)};
  padding-bottom: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const IconWrap = styled.View`
  margin-right: 12;
  flex-shrink: 0;
`;

const ContentArea = styled.View`
  flex-grow: 1;
  padding-right: 36;
  display: flex;
  align-items: flex-start;
`;

const MessageTitle = styled.Text`
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 2;
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  color: ${({ color }) => color};
`;

const MessageText = styled.Text`
  margin-bottom: 2;
  font-size: 14;
  line-height: 20;
  color: ${({ color }) => color};
`;

const ActionButton = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  max-width: 200;
  height: 36;
  margin-top: 8;
  padding-right: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const ActionIcon = styled(Icon)`
  margin-right: 24;
`;

const ActionButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  letter-spacing: -0.2;
`;

const CloseButton = styled.TouchableOpacity`
  position: absolute;
  top: 12;
  right: 12;
  color: ${({ theme }) => theme.colors.white};
`;

export {
  Wrapper,
  IconWrap,
  ContentArea,
  MessageTitle,
  MessageText,
  ActionButton,
  ActionButtonText,
  CloseButton,
  ActionIcon,
};
