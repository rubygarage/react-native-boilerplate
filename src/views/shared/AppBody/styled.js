import styled from 'styled-components/native';

const AppBody = styled.View`
  position: relative;
  flex-grow: 1;
  padding-right: 16px;
  padding-left: 16px;
  background-color: ${({ theme }) => theme.colors.black};
`;

export default AppBody;
