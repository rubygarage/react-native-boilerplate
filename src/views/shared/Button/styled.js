import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  min-width: ${(props) => (props.isSmallWidth ? 72 : 136)};
  height: ${(props) => (props.isSmallHeight ? '36px' : '48px')};
  justify-content: center;
  align-items: center;
  padding-right: 8;
  padding-left: 8;
  border-radius: 4;
  flex-shrink: 0;
  background-color: ${({ theme, isOutline, fillColor }) => (isOutline ? theme.colors.white : fillColor)};
  margin-right: ${(props) => (props.isMarginRight ? '16' : '0')};
  margin-bottom: ${(props) => (props.isMarginBottom ? '16' : '0')};
  border-width: ${(props) => (props.isBorder ? '2' : '0')};
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.black};
`;

const Text = styled.Text`
  color: ${({ theme, isOutline }) => (isOutline ? theme.colors.black : theme.colors.white)};
  font-weight: bold;
`;

export { Wrap, Text };
