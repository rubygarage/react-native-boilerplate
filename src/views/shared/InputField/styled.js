import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-direction: column;
`;

const Label = styled.Text`
  margin-bottom: 4px;
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldWrap = styled.View`
  position: relative;
`;

const Field = styled.TextInput`
  border-radius: 3px;
  min-height: 48px;
  border-width: ${({ isFocused, isError }) => (isFocused && !isError ? '2px' : '1px')};
  border-style: solid;
  border-color: ${({ theme, isError, isFocused }) => {
    if (isError) {
      return theme.colors.black;
    }
    if (isFocused) {
      return theme.colors.black;
    }
    return theme.colors.black;
  }};
  padding-top: 10px;
  padding-right: ${({ actionWidth }) => (actionWidth + 12)};
  padding-bottom: 10px;
  padding-left: 12px;
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldError = styled.View`
  flex-direction: row;
  align-items: center;
  margin-top: 6px;
  margin-bottom: 8px;
`;

const FieldErrorClose = styled.View`
  margin-right: 8px;
`;

const FieldErrorMessage = styled.Text`
  color: ${({ theme }) => theme.colors.black};
  font-size: 14px;
  line-height: 20px;
`;

const ActionButtonWrapper = styled.View`
  position: absolute;
  height: 48px;
  right: 8px;
  justify-content: center;
`;

export {
  Wrap, Label, FieldWrap, Field,
  FieldError, FieldErrorClose, FieldErrorMessage, ActionButtonWrapper,
};
