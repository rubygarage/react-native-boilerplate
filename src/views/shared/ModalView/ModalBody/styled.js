import styled from 'styled-components/native';

const Wrap = styled.View`
  padding-top: 24px;
  padding-right: 24px;
  padding-left: 24px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
