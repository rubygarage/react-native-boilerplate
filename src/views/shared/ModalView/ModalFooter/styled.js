import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-shrink: 0;
  flex-direction: row;
  justify-content: flex-end;
  padding: 16px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
