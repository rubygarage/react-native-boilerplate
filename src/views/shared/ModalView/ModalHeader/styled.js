import styled from 'styled-components/native';

const Wrap = styled.View`
  position: relative;
  flex-direction: row;
  align-items: center;
  flex-shrink: 0;
  padding: 16px 24px;
  border-bottom-color: rgba(82,97,115, 0.12);
  border-bottom-width: 1px;
  border-style: solid;
`;

const Title = styled.Text`
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.2px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.black};
`;

const Close = styled.TouchableOpacity`
  flex-shrink: 0;
  margin-right: 16px;
`;

export { Wrap, Title, Close };
