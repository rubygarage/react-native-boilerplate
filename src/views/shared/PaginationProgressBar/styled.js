import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  width: 100%;
  height: 40px;
  align-items: center;
  justify-content: center;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
