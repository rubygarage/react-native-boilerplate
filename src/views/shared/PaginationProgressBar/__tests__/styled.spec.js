import React from 'react';

import renderStyled from 'utils/testHelpers/renderStyled';
import * as S from '../styled';

describe('PaginationProgressBar styled components', () => {
  describe('S.Wrap matches snapshot', () => {
    it('default', () => {
      const container = renderStyled(<S.Wrap />);
      expect(container).toMatchSnapshot();
    });
  });
});
