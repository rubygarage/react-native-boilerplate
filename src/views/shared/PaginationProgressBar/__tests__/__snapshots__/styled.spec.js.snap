// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationProgressBar styled components S.Wrap matches snapshot default 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
/>
`;
